src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsService.kt

```diff
@@ -153,8 +154,20 @@ class AppointmentsService(
     startTime = LocalTime.parse(instance.startTime),
     endTime = LocalTime.parse(instance.endTime),
     eventSubType = instance.categoryCode,
+    comment = constructComment(instance),
   )
 
+  private fun constructComment(instance: AppointmentInstance) =
+    if (instance.appointmentDescription == null) {
+      instance.comment
+    } else {
+      if (instance.comment == null) {
+        instance.appointmentDescription
+      } else {
+        "${instance.appointmentDescription} - ${instance.comment}"
+      }
```

Comment:
Could use something like this to be idiomatic (don't know if it is any clearer!)
instance.appointmentDescription?.let{instance.comment?.let{ "${instance.appointmentDescription} - ${instance.comment}" } ?: instance.appointmentDescription} ?: instance.comment